cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManagerTest.java

```diff
@@ -0,0 +1,440 @@
+        Map<String, Double> currentQuotas = createMapOfCurrentQuotas(null, 1000L, null, 0.5);
+        QuotasPluginKafka desiredQuotasPluginKafka = new QuotasPluginKafkaBuilder()
+            .withProducerByteRate(1000L)
+            .withRequestPercentage(2)
+            .build();
+
+        List<ClientQuotaAlteration.Op> desiredQuotas = DefaultKafkaQuotasManager.prepareQuotaConfigurationRequest(desiredQuotasPluginKafka);
+
+        assertThat(DefaultKafkaQuotasManager.currentAndDesiredQuotasDiffer(currentQuotas, desiredQuotas), is(true));
+    }
+
+    /**
+     * Checks whether {@link DefaultKafkaQuotasManager#currentAndDesiredQuotasDiffer(Map, List)} is able to correctly compare
+     * the current and desired quotas in case that only one (last) quota value is different and if it correctly returns the
+     * result.
+     */
+    @Test
+    void testCurrentAndDesiredQuotasDifferWithOneDifferentQuotaValue() {
+        Long producerByteRate = 1000L;
+        Long consumerByteRate = 1000L;
+        Integer requestPercentage = 25;
+
+        Map<String, Double> currentQuotas = createMapOfCurrentQuotas(producerByteRate, consumerByteRate, requestPercentage, 0.5);
+        QuotasPluginKafka desiredQuotasPluginKafka = new QuotasPluginKafkaBuilder()
+            .withProducerByteRate(producerByteRate)
+            .withConsumerByteRate(consumerByteRate)
+            .withRequestPercentage(requestPercentage)
+            .withControllerMutationRate(0.6)
+            .build();
+
+        List<ClientQuotaAlteration.Op> desiredQuotas = DefaultKafkaQuotasManager.prepareQuotaConfigurationRequest(desiredQuotasPluginKafka);
+
+        assertThat(DefaultKafkaQuotasManager.currentAndDesiredQuotasDiffer(currentQuotas, desiredQuotas), is(true));
+    }
+
+    /**
+     * Checks if the result of {@link DefaultKafkaQuotasManager#shouldAlterDefaultQuotasConfig(Reconciliation, Vertx, Admin, List, boolean)}
+     * when there are no default Kafka quotas set and none (or Strimzi Quotas plugin) are desired.
+     * The result of this check should be false.
+     *
+     * @param context   Vertx test context
+     */
+    @Test
+    void testShouldAlterQuotasWithNoneCurrentAndNotAKafkaQuotasPlugin(VertxTestContext context) {
+        // Mock the Admin client
+        Admin mockAdminClient = mock(Admin.class);
+
+        // Mock describing the current metadata version
+        mockDescribeQuotasResultEmpty(mockAdminClient);
+
+        List<ClientQuotaAlteration.Op> noneQuotas = DefaultKafkaQuotasManager.prepareQuotaConfigurationRequest(DefaultKafkaQuotasManager.emptyQuotasPluginKafka());
+
+        Checkpoint checkpoint = context.checkpoint();
+
+        DefaultKafkaQuotasManager.shouldAlterDefaultQuotasConfig(Reconciliation.DUMMY_RECONCILIATION, vertx, mockAdminClient, noneQuotas, true)
+            .onComplete(context.succeeding(result -> {
+                context.verify(() -> assertThat(result, is(false)));
```

Comment:
Do you really need all these `context.verify(`? You seem to use it only in some places ... and I do not think we are using it in other classes.

Reply:
I saw it somewhere, but it doesn't make sense to have it just in some places. I will remove them.